opensrp-pnc/src/main/java/org/smartregister/pnc/utils/PncUtils.java

```diff
@@ -298,4 +303,77 @@ public static JSONArray generateFieldsFromJsonForm(@NonNull JSONObject jsonFormO
         return jsonArray;
     }
 
+    public static HashMap<String, HashMap<String, String>> buildRepeatingGroupTests(@NonNull JSONObject stepJsonObject, String fieldName) throws JSONException {
+        ArrayList<String> keysArrayList = new ArrayList<>();
+        JSONArray fields = stepJsonObject.optJSONArray(JsonFormConstants.FIELDS);
+        JSONObject jsonObject = JsonFormUtils.getFieldJSONObject(fields, fieldName);
+        HashMap<String, HashMap<String, String>> repeatingGroupMap = new HashMap<>();
+        if (jsonObject != null) {
+            JSONArray jsonArray = jsonObject.optJSONArray(JsonFormConstants.VALUE);
+            for (int i = 0; i < jsonArray.length(); i++) {
+                JSONObject valueField = jsonArray.optJSONObject(i);
+                String fieldKey = valueField.optString(JsonFormConstants.KEY);
+                keysArrayList.add(fieldKey);
+            }
+
+            for (int k = 0; k < fields.length(); k++) {
+                JSONObject valueField = fields.optJSONObject(k);
+                String fieldKey = valueField.optString(JsonFormConstants.KEY);
+                String fieldValue = valueField.optString(JsonFormConstants.VALUE);
+
+                if (fieldKey.contains("_")) {
+                    fieldKey = fieldKey.substring(0, fieldKey.lastIndexOf("_"));
+                    if (keysArrayList.contains(fieldKey) && StringUtils.isNotBlank(fieldValue)) {
+                        String fieldKeyId = valueField.optString(JsonFormConstants.KEY).substring(fieldKey.length() + 1);
+                        valueField.put(JsonFormConstants.KEY, fieldKey);
+                        HashMap<String, String> hashMap = repeatingGroupMap.get(fieldKeyId) == null ? new HashMap<>() : repeatingGroupMap.get(fieldKeyId);
+                        hashMap.put(fieldKey, fieldValue);
+                        repeatingGroupMap.put(fieldKeyId, hashMap);
+                    }
+                }
+            }
+        }
+        return repeatingGroupMap;
+    }
+
+    public static HashMap<String, String> getPncClient(String baseEntityId) {
+        ArrayList<HashMap<String, String>> hashMap;
+        hashMap = CoreLibrary.getInstance().context().getEventClientRepository().rawQuery(PncLibrary.getInstance().getRepository().getReadableDatabase(),
```

Comment:
you can combine declaration and assigning

Reply:
done